File: Algorithms/Sorts/QuickSort.py
from random import Random, randint


def QuickSort(arr):
    if len(arr) < 2:
        return arr
    randIndex = randint(0, len(arr) - 1)
    pivot = arr[randIndex]
    return QuickSort([ x for i, x in enumerate(arr) if x <= pivot and i != randIndex ]) + [pivot] + QuickSort([ x for i, x in enumerate(arr) if x > pivot and i != randIndex ])

# print(QuickSort([4, 9, 8, 1, 2, 0, 3, 7, 99, 23, 8]))


import random
arr = random.sample(range(100), random.randint(10, 25))
print(arr)

File: Algorithms/Sorts/QuickSort.ts
function QS(arr: number[]): number[] {
    if (arr.length < 2) return arr
    let randIndx = Math.floor(Math.random() * arr.length)
    let pivot = arr[randIndx]
    let less = arr.filter((e, i) => {
        if (i == randIndx) return false
        return e < pivot
    })
    let more = arr.filter((e, i) => {
        if (i == randIndx) return false
        return e >= pivot
    })
    return [...QS(less), pivot, ...QS(more)]
}

console.log(QS([4, 9, 8, 1, 2, 0, 3, 7, 99, 23, 8]))